src/app/components/header/header.component.html

```diff
@@ -0,0 +1,59 @@
+                        coupons</span> </div>
+            </div>
+        </div>
+    </div>
+    <div class="navbar navbar-light py-lg-4 pt-3 px-0 pb-0">
+        <div class="container">
+            <div class="row w-100 align-items-center g-lg-2 g-0">
+                <div class="col-xxl-2 col-lg-3"> <a class="navbar-brand d-none d-lg-block" routerLink="/"> <img
+                            src="/assets/images/logo/freshcart-logo.svg" alt="eCommerce HTML Template"> </a>
+                    <div class="d-flex justify-content-between w-100 d-lg-none"> <a class="navbar-brand" routerLink="/">
+                            <img src="/assets/images/logo/freshcart-logo.svg" alt="eCommerce HTML Template"> </a>
+                        <div class="d-flex align-items-center lh-1">
+                            <button class="navbar-toggler collapsed pull-left" type="button" data-bs-toggle="collapse"
+                                data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
+                                aria-expanded="false" aria-label="Toggle navigation"> <span
+                                    class="icon-bar top-bar mt-0"></span> <span class="icon-bar middle-bar"></span>
+                                <span class="icon-bar bottom-bar"></span>
+                            </button>
+                        </div>
+                    </div>
+                    <div class="navbar-collapse collapse" id="navbarSupportedContent">
+                        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+                            <li *ngFor="let category of categories$ | async as categoryList" class="nav-item">
+                                <a class="nav-link active" aria-current="page"
+                                    routerLink="categories/{{category.id}}">{{category.name}}</a>
+                            </li>
+                        </ul>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="border-bottom pb-lg-4 pb-3">
+        <nav class="navbar-collapse navbar navbar-expand-lg navbar-light navbar-default pt-0 pb-0">
+            <div class="container px-0 px-md-3">
+                <div class="offcanvas offcanvas-start p-4 p-lg-0">
+                    <div class="d-flex justify-content-between align-items-center mb-2 d-block d-lg-none"> <a
+                            routerLink="/"><img src="/assets/images/logo/freshcart-logo.svg"
+                                alt="eCommerce HTML Template"></a> <button type="button" class="btn-close"
+                            aria-label="Close"></button>
+                    </div>
+                    <div class="d-block">
+                        <ul class="nav col-12 col-lg-auto me-lg-auto mb-2 mb-md-0">
+                            <li *ngFor="let category of categories$ | async as categoryList"><a
```

Comment:
With "as categoryList" as above.

You subscribe to categories$ twice. You should do it only once

Reply:
Fixed it by subscribing to the categories in heder tag,